File: lwjgl3/src/main/java/io/github/msameer0/rhythmicrush/lwjgl3/atlases/PackLevelSelectAtlas.java
package io.github.msameer0.rhythmicrush.lwjgl3.atlases;

import com.badlogic.gdx.tools.texturepacker.TexturePacker;

public class PackLevelSelectAtlas {
    public static void main(String[] args) {

        TexturePacker.Settings settings = new TexturePacker.Settings();
        settings.maxWidth = 2048;       // max page width
        settings.maxHeight = 2048;      // max page height
        settings.edgePadding = true;    // padding around images
        settings.duplicatePadding = true;
        settings.scale = new float[]{0.65f};         // scale all images to 65% of original size

        TexturePacker.process(settings,
            "assets/level_select", // input folder
            "assets/level_select_atlases",              // output folder
            "level_select"         // atlas name
        );

        System.out.println("Level select atlas packed!");
    }
}

File: lwjgl3/src/main/kotlin/io/github/msameer0/rhythmicrush/lwjgl3/atlases/PackLevelSelectAtlas.kt
package io.github.msameer0.rhythmicrush.lwjgl3.atlases

import com.badlogic.gdx.tools.texturepacker.TexturePacker

object PackLevelSelectAtlas {
    @JvmStatic
    fun main(args: Array<String>) {
        val settings = TexturePacker.Settings()
        settings.maxWidth = 2048
        settings.maxHeight = 2048
        settings.edgePadding = true
        settings.duplicatePadding = true
        // Base scale for level select was 0.65f, but we'll use 1.0f as high quality for consistency now,
        // or we could multiply our quality scales by 0.65f.
        // Given the request "pack thrice the amount, treat current as high", I'll stick to 1.0, 0.5, 0.25.

        AtlasPackingUtils.packAllQualities(
            "textures_to_put_in_atlases/level_select",
            "assets/level_select_atlases",
            "level_select",
            settings
        )
    }
}
